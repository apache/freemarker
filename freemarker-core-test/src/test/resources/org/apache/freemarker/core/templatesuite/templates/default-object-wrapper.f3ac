<#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
<#list array as item>
${item}
</#list>
${array?size}
${array[0]}
${array[1]}
<#list list as item>
${item}
</#list>
${list?size}
${list?hasContent?then('not empty', 'empty')}
${list[0]}
${map.key}
${map?api.get(objKey)}
${obj.foo}
<#if obj.foo??>hasfoo<#else>nofoo</#if>
<#if obj.baz??>hasbaz<#else>nobaz</#if>
${obj.bar[0]}
${obj.getFoo()}
${obj.overloaded(1?int)}
${obj.overloaded("String")}
${resourceBundle.message}
${resourceBundle("format", date)}
<#assign static = statics["org.apache.freemarker.core.templatesuite.models.BeanTestClass"]>
${static.staticMethod()}
${static.staticOverloaded(1)}
${static.staticOverloaded("String")}
${static.STATIC_FINAL_FIELD}
${static.STATIC_FIELD}
<#assign enum = enums["org.apache.freemarker.core.templatesuite.models.EnumTestClass"]>
${enum.ONE}
${enum.TWO}
${enum.THREE}
${(enum.ONE == enum.ONE)?string("true", "false")}
${(enum.ONE == enum.TWO)?string("true", "false")}
${enums["org.apache.freemarker.core.templatesuite.models.BeanTestClass"]???string("true", "false")}
${obj.something}
${obj.publicInner.x}
${obj.publicInner.m()}
<@assertFails message="obj.privateInner.x">${obj.privateInner.x}</@>
<@assertFails message="obj.privateInner.m">${obj.privateInner.m()}</@>