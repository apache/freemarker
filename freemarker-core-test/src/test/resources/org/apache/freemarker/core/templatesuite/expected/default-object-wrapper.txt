/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
array-0
array-1
2
array-0
array-1
list-0
list-1
list-2
3
not empty
list-0
value
objValue
foo-value
hasfoo
nobaz
bar-value-0
foo-value
overloaded-int-1
overloaded-String-String
Message
1974-11-14
static-method
static-overloaded-int-1
static-overloaded-String-String
static-final-field
static-field
ONEx
TWOx
THREEx
true
false
false
42
1
m

